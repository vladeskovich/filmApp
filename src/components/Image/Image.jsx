import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './Image.scss';

const Image = ({
  className,
  src,
}) => {
  const classes = classNames(styles.image, className);

  return (
    <img
      src={src}
      className={classes}
    />
  );
};

Image.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
};

export default Image;
