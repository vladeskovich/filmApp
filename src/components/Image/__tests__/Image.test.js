import React from 'react';
import Image from '../Image';

describe('Image', () => {
  const props = {
    src: 'assets/image',
    className: 'buttonTest',
  };

  it('should render Image component without props', () => {
    const component = shallow(<Image/>);

    expect(component).toMatchSnapshot();
  });

  it('should render Image component with props', () => {
    const component = shallow(<Image {...props}/>);

    expect(component).toMatchSnapshot();
  });
});
