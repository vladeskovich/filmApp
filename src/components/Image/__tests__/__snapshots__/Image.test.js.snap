// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render Image component with props 1`] = `
<img
  className="Image__image_2mBUrUSU buttonTest"
  src="assets/image"
/>
`;

exports[`Image should render Image component without props 1`] = `
<img
  className="Image__image_2mBUrUSU"
/>
`;
